### PR TITLE
Allow shortenedExport() to use __toString() on objects if defined

### PR DESCRIPTION
**Pull request by theseer**
_This PR implements this and is meant as a starting point for discussion._

Not necessarily the final implementation yet but it would be quite handy to have shortenedExport() not only show ... for object properties if the class in question implements __toString.

I didn't add a test for it yet as I'm not sure where to place the code for the dummy class that would implement __toString()?

### CODE REVIEW

File: src/Exporter.php [hunk 1]
```diff
@@ -120,7 +120,7 @@ public function shortenedExport($value)
             return sprintf(
                 '%s Object (%s)',
                 get_class($value),
-                count($this->toArray($value)) > 0 ? '...' : ''
+                method_exists($value, '__toString') ? (string)$value : (count($this->toArray($value)) > 0 ? '...' : '')
```
Comment (GrahamCampbell) [inline src/Exporter.php]: missing space in `(string)$value`

Comment (theseer) [inline src/Exporter.php]: You realize that this is a DISCUSSION PR about a *feature*
Commenting on *formatting* is NOT helping.

Comment (sebastianbergmann) [inline src/Exporter.php]: I have to agree with @theseer, @GrahamCampbell. This is the same behavior that got @ircmaxell upset in https://github.com/sebastianbergmann/phpunit/pull/1793#issuecomment-121775187

Comment (GrahamCampbell) [inline src/Exporter.php]: Sorry. Didn't notice the PR author. Wouldn't have commented if I'd seen it was you. :)

Comment (GrahamCampbell) [inline src/Exporter.php]: I'm just so used to reviewing laravel PRs.

Comment (ircmaxell) [inline src/Exporter.php]: Most Important Part

Comment (GrahamCampbell) [inline src/Exporter.php]: LOL. So true. :)

**Comment by keradus** [comment]
idea behind this PR is nice ;)

**Comment by bigaltum** [comment]
``` php
        if (is_object($value)) {
            if (method_exists($value, '__toString')) {
                return (string) $value;
            } else {
                return sprintf(
                    '%s Object (%s)',
                    get_class($value),
                    count($this->toArray($value)) > 0 ? '...' : ''
                );
            }
        }
```

I think it's better.

**Comment by sebastianbergmann** [comment]
Thank you for your contribution. I appreciate the time you invested in preparing this pull request. However, I have decided not to merge it.

**Pull request closed without merging**